src/helpers/proofs/fetchCentralizedProof.ts

```diff
@@ -12,9 +13,20 @@ export default async function (
   let result: ECDSAProofStruct = {} as ECDSAProofStruct
 
   while (!Object.keys(result).length) {
-    const { status, result: jobResult } = await sendRequest(id, proverAddress)
-    if (status === JobStatus.completed) result = makeTransaction(jobResult)
+    const {
+      status,
+      position,
+      result: jobResult,
+    } = await sendRequest(id, proverAddress)
+    if (status === JobStatus.completed) {
+      result = makeTransaction(jobResult)
+      JobStore.queuePosition = undefined
+    }
+    if (status === JobStatus.scheduled || status === JobStatus.running) {
```

Comment:
We need to check for running as well, because if the server will send `status = 'running'`, but we won't check it, the position will stay at `1`, but it's `0` actually and for zero we don't display text